src/chainstate/stacks/miner.rs

```diff
@@ -761,23 +760,44 @@ impl StacksBlockBuilder {
     /// Finish building the anchored block.
     /// TODO: expand to deny mining a block whose anchored static checks fail (and allow the caller
     /// to disable this, in order to test mining invalid blocks)
-    pub fn mine_anchored_block<'a>(&mut self, clarity_tx: &mut ClarityTx<'a>) -> StacksBlock {
+    pub fn mine_anchored_block(&mut self, clarity_tx: &mut ClarityTx) -> StacksBlock {
         assert!(!self.anchored_done);
 
         // add miner payments
         if let Some((ref miner_reward, ref user_rewards, ref parent_reward)) = self.miner_payouts {
             // grant in order by miner, then users
-            StacksChainState::process_matured_miner_rewards(
+            let matured_ustx = StacksChainState::process_matured_miner_rewards(
                 clarity_tx,
                 miner_reward,
                 user_rewards,
                 parent_reward,
             )
             .expect("FATAL: failed to process miner rewards");
+
+            clarity_tx
```

Comment:
I've seen this code block a few times.  Should `clarity_tx` just have its own method for incrementing the total uSTX supply?